packages/rocket-file-uploads-core/src/index.ts

```diff
@@ -1,27 +1,49 @@
 import { BoosterConfig, RocketDescriptor } from '@boostercloud/framework-types'
-import { functionID, RocketFilesParams } from '@boostercloud/rocket-file-uploads-types'
+import {
+  functionID,
+  RocketFilesConfiguration,
+  RocketFilesUserConfiguration,
+  RocketProviderPackageType,
+} from '@boostercloud/rocket-file-uploads-types'
 import { fileUploaded } from './file-uploaded'
 
 export { FileHandler } from './file-handler'
 
 export class BoosterRocketFiles {
-  public constructor(readonly config: BoosterConfig, readonly params: RocketFilesParams) {
-    config.registerRocketFunction(functionID, async (config: BoosterConfig, request: unknown) => {
-      return fileUploaded(config, request, params)
-    })
-  }
-
-  public rocketForAzure(): RocketDescriptor {
+  public rocketForAzure(config: BoosterConfig, userConfiguration: RocketFilesUserConfiguration): RocketDescriptor {
+    const configuration = BoosterRocketFiles.buildParameters(
+      userConfiguration,
+      '@boostercloud/rocket-file-uploads-azure'
+    )
+    this.register(config, configuration)
     return {
       packageName: '@boostercloud/rocket-file-uploads-azure-infrastructure',
-      parameters: this.params,
+      parameters: configuration,
     }
   }
 
-  public rocketForLocal(): RocketDescriptor {
+  public rocketForLocal(config: BoosterConfig, userConfiguration: RocketFilesUserConfiguration): RocketDescriptor {
+    const configuration = BoosterRocketFiles.buildParameters(
+      userConfiguration,
+      '@boostercloud/rocket-file-uploads-local'
+    )
+    this.register(config, configuration)
     return {
       packageName: '@boostercloud/rocket-file-uploads-local-infrastructure',
-      parameters: this.params,
+      parameters: configuration,
     }
   }
+
+  private register(config: BoosterConfig, configuration: RocketFilesConfiguration): void {
+    config.registerRocketFunction(functionID, async (config: BoosterConfig, request: unknown) => {
+      return fileUploaded(config, request, configuration)
+    })
+  }
```

Comment:
And here take the config from `this`
```suggestion
  private register(configuration: RocketFilesConfiguration): void {
    this.config.registerRocketFunction(functionID, async (config: BoosterConfig, request: unknown) => {
      return fileUploaded(this.config, request, configuration)
    })
  }
```